api/src/test/java/org/openmrs/api/AdministrationServiceTest.java

```diff
@@ -927,4 +928,59 @@ public void getPresentationLocales_shouldReturnLanguageLocaleIfItIsSpecifiedInAl
 		Assert.assertTrue("en", presentationLocales.contains(new Locale("en")));
 		Assert.assertTrue("es", presentationLocales.contains(new Locale("es")));
 	}
+	
+	/**
+	 * @see AdministrationService#getPresentationLocales()
+	 */
+	@Test
+	@Verifies(value = "should preserve insertion in Set returned by method", method = "getPresentationLocales()")
+	public void getPresentationLocales_shouldPreserveInsertionInSetReturnedByMethod()
+			throws Exception {
+		String globalPropertyLocaleListAllowedData = "en_GB, es, ja_JP, it_IT, pl_PL";
+		//The order of languages and locales is described above and should be followed bt `presentationLocales` Set
+		Context.getAdministrationService().saveGlobalProperty(
+				new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_LOCALE_ALLOWED_LIST, globalPropertyLocaleListAllowedData));
+		
+		List<Locale> locales = new ArrayList<Locale>();
+		//Add data in random order and verify that order is maintained in the end by checking against order in global property
+		locales.add(new Locale("pl", "PL"));
+		locales.add(new Locale("es"));
+		locales.add(new Locale("en"));
+		locales.add(new Locale("it", "IT"));
+		
+		MutableResourceBundleMessageSource mutableResourceBundleMessageSource = Mockito
+				.mock(MutableResourceBundleMessageSource.class);
+		Mockito.when(mutableResourceBundleMessageSource.getLocales()).thenReturn(locales);
```

Comment:
Ok but the above template is followed in all the other test methods. Wouldn't you prefer I keep it that way? I tried to emulate the pattern of the other tests as much as I could since I figured they would be they way someone wanted to for a reason. Should I let it be then?